contracts/NFTStorefrontV2.cdc

```diff
@@ -293,6 +298,21 @@ pub contract NFTStorefrontV2 {
             return self.marketplacesCapability
         }
 
+        /// hasNFTPresentInListingProvider
+        /// Tells whether listed NFT is present in provided capability.
+        /// If it returns `false` then it means listing becomes ghost or sold out.
+        pub fun hasNFTPresentInListingProvider(): Bool {
+            if let providerRef = self.nftProviderCapability.borrow() {
+                let availableIDs = providerRef.getIDs()
+                for id in availableIDs {
```

Comment:
np: not sure if `availableIDs.contains(self.details.nftID)` is more efficient than a for loop or if it's all the same

Reply:
I think you are right, Maybe I was dreaming that time to use the for loop for the same. Will fix it. Although it should not create any different in terms of computation.